rtslib/root.py

```diff
@@ -327,30 +332,32 @@ def err_func(err_str):
             if 'name' not in so:
                 err_func("'name' not defined in storage object %d" % index)
                 continue
-            try:
-                so_cls = so_mapping[so['plugin']]
-            except KeyError:
-                err_func("'plugin' not defined or invalid in storageobject %s" % so['name'])
-                continue
-            kwargs = so.copy()
-            dict_remove(kwargs, ('exists', 'attributes', 'plugin', 'buffered_mode', 'alua_tpgs'))
-            try:
-                so_obj = so_cls(**kwargs)
-            except Exception as e:
-                err_func("Could not create StorageObject %s: %s" % (so['name'], e))
-                continue
-
-            # Custom err func to include block name
-            def so_err_func(x):
-                return err_func("Storage Object %s/%s: %s" % (so['plugin'], so['name'], x))
-
-            set_attributes(so_obj, so.get('attributes', {}), so_err_func)
 
-            for alua_tpg in so.get('alua_tpgs', {}):
-               try:
-                   ALUATargetPortGroup.setup(so_obj, alua_tpg, err_func)
-               except RTSLibALUANotSupported:
-                   pass
+            if (not storage_object and not target) or (storage_object and so['name'] == storage_object):
+                try:
```

Comment:
Maybe in this loop if the storage_object != None and once we have found it from the config file, we should break the loop directly no matter after all the restore options are done successfully or failed, since it not allows to have the same 'name' in the same 'plugin' and it makes no sense to continue the rest. right ?

Reply:
Yeah, good catch, should have break'ed.